draft-ietf-dnsop-svcb-https.md

```diff
@@ -1702,6 +1702,244 @@ the authoritative.  Thus, splitting the RR type would double, or in
+    \x03foo\x07example\x03org\x00                      # target
+    \x00\x00                                           # key 0
+    \x00\x02                                           # param length 2
+    \x00\x01                                           # value: key 1
+    \x00\x01                                           # key 1
+    \x00\x09                                           # param length 9
+    \x02                                               # alpn length 2
+    h2                                                 # alpn value
+    \x05                                               # alpn length 5
+    h3-19                                              # alpn value
+    \x00\x04                                           # key 4
+    \x00\x04                                           # param length 4
+    \xc0\x00\x02\x01                                   # param value
+
+This last vector has an alpn value with an escaped comma and an escaped
+backslash in two presentation formats.
+
+    16 foo.example.org alpn="f\\\\oo\\,bar,h2"
+    16 foo.example.org alpn=f\\\092oo\092,bar,h2
+
+    \# 35 (
+    00 10                                              ; priority
+    03 66 6f 6f 07 65 78 61 6d 70 6c 65 03 6f 72 67 00 ; target
+    00 01                                              ; key 1
+    00 0c                                              ; param length 12
+    08                                                 ; alpn length 8
+    66 5c 6f 6f 2c 62 61 72                            ; alpn value
+    02                                                 ; alpn length 2
+    68 32                                              ; alpn value
+    )
+
+    \x00\x10                                           # priority
+    \x03foo\x07example\x03org\x00                      # target
+    \x00\x01                                           # key 1
+    \x00\x0c                                           # param length 12
+    \x08                                               # alpn length 8
+    f\oo,bar                                           # alpn value
+    \x02                                               # alpn length 2
+    h2                                                 # alpn value
+
+
+## Failure cases
+
+In this subsection, example resource records are shown which are not
+compliant with this document. The various reasons for non-compliance
+are explained with each example.
+
+
+This example has multiple instances of the same
+SvcParamKey {{presentation}}.
+
+    example.com.   SVCB   1 foo.example.com. (
+                           key123=abc key123=def
+                           )
+
+In the next examples the SvcParamKeys are missing their values.
+
+    example.com.   SVCB   1 foo.example.com. mandatory
+    example.com.   SVCB   1 foo.example.com. alpn
+    example.com.   SVCB   1 foo.example.com. port
+    example.com.   SVCB   1 foo.example.com. ipv4hint
+    example.com.   SVCB   1 foo.example.com. echconfig
```

Comment:
This is a tricky one.  The param indicates a valid SVCB `echconfig` parameter whose value is "" (which is what you get when you base64-decode "").

Of course, "" is currently not a valid ECHConfigList, and is unlikely to be, but the ECH specification is not final.  In any event, I don't think there's an obligation for zone file parsers to check whether the ECHConfigList is valid, which may be difficult if they're not linked to an ECH-capable TLS library.

Reply:
Removed that one.